server/src/main/java/org/opensearch/common/cache/store/StoreAwareCacheValue.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.store;
+
+import org.opensearch.common.cache.store.enums.CacheStoreType;
+
+/**
+ * Represents a store aware cache value.
+ * @param <V> Type of value.
+ *
+ * @opensearch.internal
+ */
+public class StoreAwareCacheValue<V> {
```

Comment:
StoreAwareCache already is aware about the tier/store and the consumer shouldn't care about where is the value coming from, wondering why do we need this?

Reply:
You can provide the concrete implementation of the specific listeners of each cache type and wire them up as a part of the store aware cache. I am trying to see if firing events could be delegated to individual caches directly

Reply:
In my previous revisions, I did that ie firing events from individual caches directly and then aggregation those in overall TieredCache. It became hard/impossible to do that when changes were made in tieredCache.computeIfAbsent() to handle specific scenarios. Had to resolve to then TieredCache handling its own listener logic.